src/server.js

```diff
@@ -0,0 +1,10 @@
+const app = require('./app.js');
+require('env2')('config.env');
+
+app.set('PORT', process.env.PORT);
+app.listen(app.get('PORT'), () => {
+  // eslint-disable-next-line no-console
+  console.log(`The Server Is Up And Running On http://localhost:${app.get('PORT')}`);
+  // eslint-disable-next-line global-require
+  require('./mongoose.config');
```

Comment:
at the moment you are running your server, and then connecting to mongoose/mongo within the server. I would do this the other way around/restructure this `mongo.config` file:

```js
// mongo.config.js
const mongoose = require('mongoose');
require('env2')('config.env');

const mongoConnection = mongoose.connect(process.env.DATABASE_URL, { useNewUrlParser: true, useCreateIndex: true });

mongoConnection.connection.on('error', (err) => {
  // eslint-disable-next-line no-console
  console.log('mongoose connection error: ', err);
});
mongoConnection.connection.on('disconnected', () => {
  // eslint-disable-next-line no-console
  console.log('mongoose has been disconnected!');
});

module.exports = mongoConnection
```

```js
// server.js
const app = require('./app.js');
const db = require('./mongoose.config.js')
require('env2')('config.env');

db.connection.on('connected', () => {
  app.listen(app.get('PORT'), () => {
  // eslint-disable-next-line no-console
  console.log(`The Server Is Up And Running On http://localhost:${app.get('PORT')}`);
  })
});
```

Does this make sense?

See here as well: https://stackoverflow.com/questions/24474386/pass-mongoose-connection-to-module